src/main/java/com/synopsys/integration/alert/channel/jira/cloud/JiraCloudContextBuilder.java

```diff
@@ -22,13 +22,21 @@
  */
 package com.synopsys.integration.alert.channel.jira.cloud;
 
-import com.synopsys.integration.alert.channel.jira.cloud.JiraCloudContext;
-import com.synopsys.integration.alert.channel.jira.cloud.JiraCloudProperties;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
 import com.synopsys.integration.alert.channel.jira.cloud.descriptor.JiraCloudDescriptor;
 import com.synopsys.integration.alert.channel.jira.common.JiraContextBuilder;
 import com.synopsys.integration.alert.common.persistence.accessor.FieldAccessor;
+import com.synopsys.integration.alert.common.rest.ProxyManager;
 
+@Component
 public class JiraCloudContextBuilder extends JiraContextBuilder<JiraCloudContext> {
```

Comment:
What are the implications of making this a component? Why wasn't it a component before?

Reply:
This wasn't a component before because it had no state and no dependencies. Now it requires a ProxyManager. So rather than autowiring in ProxyManager in the classes this class is used in and passing it in when constructing this class, we can now just have this class be autowired in where it is used.